lab-david/route/photoalbum-router.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const createError = require('http-errors');
+const debug = require('debug')('cfgram:photoalbum-router');
+
+const PhotoAlbum = require('../model/photoalbum.js');
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+
+const photoAlbumRouter = module.exports = Router();
+
+photoAlbumRouter.post('/api/photoalbum', bearerAuth, jsonParser, (req, res, next) => {
+  debug('POST: /api/photoalbum');
+
+
+  req.body.userId = req.user._id;
+  new PhotoAlbum(req.body).save()
+    .then( photoAlbum => res.json(photoAlbum))
+    .catch(err => {
+      err = createError(404, err.message);
+      next(err);
+    });
+});
+
+photoAlbumRouter.get('/api/photoalbum/:photoalbumId', bearerAuth, jsonParser, (req, res, next) => {
+  debug('GET: /api/photoalbum/:photoalbumId');
+
+  PhotoAlbum.findById(req.params.photoalbumId)
+    .then(photoAlbum => res.json(photoAlbum))
```

Comment:
You should add a check in here to ensure that you actually found a photoalbum, otherwise you could be sending back null